src/LanguageServer.Engine/LanguageServerModule.cs

```diff
@@ -124,18 +124,21 @@ static void addRegistrations(IServiceCollection services, IComponentContext comp
                             // Register all completion providers.
                             addRegistrations(services, currentScope, false, typeof(ICompletionProvider), typeof(CompletionProvider));
                         })
-                        .OnInitialize((languageServer, initializationParameters) =>
+                        .WithHandler<DocumentSyncHandler>()
+                        .WithHandler<ConfigurationHandler>()
+                        .WithHandler<DocumentSymbolHandler>()
+                        .WithHandler<DefinitionHandler>()
+                        .WithHandler<HoverHandler>()
+                        .WithHandler<CompletionHandler>()
+                        .OnInitialize((languageServer, initializationParameters, ctx) =>
                         {
-                            // Register all handlers. Now possible inside OnInitialize since v0.11.1 of OmniSharp LSP libs.
-                            languageServer.AddHandlers(currentScope.Resolve<IEnumerable<Handler>>().ToArray());
-
                             var configurationHandler = currentScope.Resolve<ConfigurationHandler>();
                             configurationHandler.Configuration.UpdateFrom(initializationParameters);
 
                             return Task.CompletedTask;
                         })
                         // TODO: remove this workaround once we upgrade to omniSharp LSP libs v0.17.0 or newer
-                        .OnStarted((server, initializationResult) =>
+                        .OnStarted((server, initializationResult, ctx) =>
```

Comment:
```suggestion
                        .OnStarted((server, initializationResult, cancellationToken) =>
```
The 3rd parameter is no "context" object, it should be renamed to "cancellationToken" to make it clear?